packages/ui/src/components/evm/erc20-token-picker/hooks/useSearchedTokens.ts

```diff
@@ -1,63 +1,24 @@
 import { useMemo } from "react";
-import { BigNumber } from "@ethersproject/bignumber";
-import { Address, erc20ABI, useContractReads } from "wagmi";
-import {
-    cachedTokenInfoWithBalanceInChain,
-    filterERC20Tokens,
-    sortERC20Tokens,
-} from "../../../../utils/erc20";
+import { filterERC20Tokens, sortERC20Tokens } from "../../../../utils/erc20";
 import { TokenInfoWithBalance, TokenListWithBalance } from "../types";
 
 export const useSearchedTokens = (
     debouncedQuery?: string,
     chainId?: number,
-    selectedList?: TokenListWithBalance,
-    withBalances?: boolean,
-    connectedAccountAddress?: string
-): { tokens: TokenInfoWithBalance[]; loadingBalances: boolean } => {
-    const cachedImportedTokens = cachedTokenInfoWithBalanceInChain(chainId);
```

Comment:
Why did we remove so much stuff from here? The cached tokens were useful, otherwise the tokens imported by the user won't be persisted, and will be removed each time the user closes the modal.

Reply:
The caching functionality has been moved to the erc20 token template, where the `erc20-token-picker` is used. I removed it because the cached token logic, when caching an imported token, relied on `wagmi` (the `useImportableToken` that fetched the token info).

Now the cache it's implemented outside of the component. If we feel we're removing too much from the UI component, I can try to think of another way to tackle this.

Reply:
Let's just leave it like this for now I'd say